editor/components/block-list/index.js

```diff
@@ -62,15 +70,45 @@ class BlockList extends Component {
 		document.addEventListener( 'copy', this.onCopy );
 		document.addEventListener( 'cut', this.onCut );
 		window.addEventListener( 'mousemove', this.setLastClientY );
+		// Needs document to also work in inspector.
+		// In other words, if there is selection, but no focus.
+		document.addEventListener( 'keydown', this.onKeyDown );
 	}
 
 	componentWillUnmount() {
 		document.removeEventListener( 'copy', this.onCopy );
 		document.removeEventListener( 'cut', this.onCut );
 		window.removeEventListener( 'mousemove', this.setLastClientY );
+		document.removeEventListener( 'keydown', this.onKeyDown );
 	}
 
 	componentWillReceiveProps( nextProps ) {
+		const prevCommonName = this.commonName;
+
+		if ( nextProps.selectedBlock ) {
+			this.blocks = [ nextProps.selectedBlock ];
+			this.commonName = nextProps.selectedBlock.name;
+		} else if ( nextProps.multiSelectedBlocks.length ) {
+			this.blocks = nextProps.multiSelectedBlocks;
+
+			const firstName = first( nextProps.multiSelectedBlocks ).name;
+
+			if ( every( nextProps.multiSelectedBlocks, ( { name } ) => name === firstName ) ) {
+				this.commonName = firstName;
+			} else {
+				delete this.commonName;
+			}
```

Comment:
What if we want to right-align a selection of paragraphs and images?

Reply:
Okay, this is getting complicated. :) Transforms on different block types have not been done before, but it doesn't sound like a bad idea in some cases. In the case of paragraphs and images though, it means something different (align vs text-align).